### PR TITLE
Enrich content 

### PR DESCRIPTION
**Pull request by piclane**
Enrich posting content with JavaScript support.
Please read README.md m(_ _)m

### CODE REVIEW

File: src/main/java/jp/co/dwango/logback/SlackWebhookAppender.java [hunk 1]
```diff
@@ -1,167 +1,434 @@
...
+        // looks up and creates JavaScript engine
+        this.engine = SCRIPT_MANAGER.getEngineByName("js");
+        
+        // bind hostname
+        Bindings bindings = engine.getBindings(ScriptContext.ENGINE_SCOPE);
+        bindings.put("hostname", resolveHostname());
+        
+        // define formatTimestamp function
+        engine.eval(
+            "function formatTimestamp(timestamp) { " +
```
Comment (tayama0324) [inline src/main/java/jp/co/dwango/logback/SlackWebhookAppender.java:182]: You can format timestamp by specifying `<pattern>%date{ISO8601}</pattern>` to your logback.xml.

File: src/main/java/jp/co/dwango/logback/SlackWebhookAppender.java [hunk 2]
```diff
@@ -1,167 +1,434 @@
...
-    private String createText(ILoggingEvent event) {
-        Layout<ILoggingEvent> layout = this.layout;
-        if (layout == null) {
-            return event.getFormattedMessage();
+    /**
+     * Resolves hostname
+     * 
+     * @return hostname
+     */
+    private static String resolveHostname() {
```
Comment (tayama0324) [inline src/main/java/jp/co/dwango/logback/SlackWebhookAppender.java:269]: You can refer `$HOSTNAME` variable in your logback.xml.

File: README.md [hunk 3]
```diff
@@ -25,6 +25,154 @@ Logback appender which posts logs to slack via incoming webhook.
...
+      "short": false
+    }]
+  }]
+}
+```
+
+##### Payload sample - Complex
+
+```javascript
+var env = 'dev', mention = '';
```
Comment (tayama0324) [inline README.md:139]: I do not like to write some logic in payload. Why not using different logback.xml on each environment? If you really want to use conditional expression in a single xml file, you can use [if statement](https://logback.qos.ch/manual/configuration_ja.html#conditional) as well.

**Review by tayama0324 — COMMENTED** [review]
Mmm... Many things you want to do here seems to be achieved by logback's basic functionality (examples are denoted inline). I think we just want to add `attatchments` property to the legacy appender, and it would be more efficient and flexible. I will try to refactor later on.

**Comment by piclane** [comment]
I know that we can build payload to some extent with only standard logback feature. And I also agree that it is not highly recommended to write logic in logback.xml.
But many customizations are required to display the log to Slack. I think that JavaScript is more customizable and easier to handle than the standard logback feature. Also, since we send JSON to Slack, it was better to treat it as JavaScript rather than handling payload as a string.
It certainly may not have been necessary to make it customizable outside the “attachments” node, but I think that this strategy is the best.

**Pull request merged**